middlewares/hacker.middleware.js

```diff
@@ -241,6 +270,42 @@ async function updateStatusIfApplicationCompleted(req, res, next) {
     }
 }
 
+/**
+ * Checks that the hacker's status matches one of the input statuses
+ * @param {String[]} statuses
+ * @returns {(req, res, next) => {}} the middleware that will check hacker's status
+ */
+function checkStatus(statuses) {
+    return Middleware.Util.asyncMiddleware(async (req, res, next) => {
+
+        let hacker = await Services.Hacker.findById(req.params.id);
+
+        if (!!hacker) {
+            const status = hacker.status;
+            // makes sure the hacker's status is in the accepted statuses list
+            if (statuses.indexOf(status) === -1) {
+                return next({
+                    status: 409,
+                    message: Constants.Error.HACKER_CHECKIN_409_MESSAGE,
```

Comment:
Should not be CHECKIN_409_MESSAGE.

Reply:
We are checking that the state of hacker.status is acceptable. Would you prefer that we use status 422, and I change HACKER_CHECKIN_409_MESSAGE in constants to HACKER_CHECKIN_422? Otherwise I'm not sure what you mean.

Reply:
This is a general `checkStatus` message-- the array `statuses` may not even have to do with checking in a hacker (for example, we might want to use this method to verify that a hacker has submitted their application). A more appropriate message might be something like `HACKER_STATUS_409_MESSAGE (Invalid status)` or something, and the data contains the acceptable statuses. Do you know what I mean?

Reply:
True, I didn't think about that.